pkg/planner/core/plan_cache_test.go

```diff
@@ -852,16 +852,22 @@ func testRandomPlanCacheCases(t *testing.T,
 	tk := testkit.NewTestKit(t, store)
 	prepFunc(tk)
 
-	// prepared plan cache
+	// nonprepared plan cache
 	for _, q := range queryFunc(true) {
 		tk.MustExec("set tidb_enable_non_prepared_plan_cache=0")
 		result1 := tk.MustQuery(q).Sort()
 		tk.MustExec("set tidb_enable_non_prepared_plan_cache=1")
+
+		// the first execution caches the plan
 		result2 := tk.MustQuery(q).Sort()
```

Comment:
If you want, you can also verify that the plan was not from the cache:
```suggestion
		result2 := tk.MustQuery(q).Sort()
		tk.MustQuery(`select @@last_plan_from_cache`).Check(testkit.Rows("0"))
```

Reply:
Some of the queries actually hit the cache.  I will remove this comments.